File: server/service/agent/player.lua
local skynet = require "skynet"

local Player = {}

Player.__index = Player

function Player.new()
	local o = {}
	setmetatable(o, Player)
	return o
end

-- 数据库加载玩家数据
function Player:load(conf)
	self.url = conf.url
	self.id = conf.id
	self.account = conf.account
	self.name = conf.name
	self.agent = skynet.self()
	self.money = conf.money
	self.usertype = conf.usertype
	self.fd = conf.fd
	self.sex = conf.sex
end

function Player:rejoin()
	local info = {
		tab_id = msg.tab_id,
		id = self.id,
		agent = skynet.self(),
		name = player.name,
		headurl = player.url,
		fd = player.fd
	}
	local resp = skynet.call(env.room, "lua", "rejoin", info)
	if resp then
		env.send_msg(resp.name, resp.msg)
	end

end

return Player

File: client/Assets/Scripts/role/Player.cs
using UnityEngine;
using System.Collections;

public class Player{

    private int _id;
    private string _name;
    private int _pos;
    private string _url;
    private bool _isReady = false;
    private bool _sex = true;

    public Player(int id,string name,int pos,int sex)
    {
        Id = id;
        Name = name;
        Pos = pos;
        Sex = (sex==1?true:false);  
    }
    public Player()
    {

    }
    public int Id
    {
        get
        {
            return _id;
        }

        set
        {
            _id = value;
        }
    }

    public string Name
    {
        get
        {
            return _name;
        }

        set
        {
            _name = value;
        }
    }

    public int Pos
    {
        get
        {
            return _pos;
        }

        set
        {
            _pos = value;
        }
    }

    public bool IsReady
    {
        get
        {
            return _isReady;
        }

        set
        {
            _isReady = value;
        }
    }

    public string Url
    {
        get
        {
            return _url;
        }

        set
        {
            _url = value;
        }
    }

    public bool Sex
    {
        get
        {
            return _sex;
        }

        set
        {
            _sex = value;
        }
    }
}
